data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java

```diff
@@ -73,12 +81,22 @@ AuthenticationResult acquireAccessTokenUsingDeviceCodeFlow() throws Exception {
         return result;
     }
 
-    String acquireAccessToken() throws DataServiceException, DataClientException {
-        if (clientCredential != null) {
-            return acquireAadApplicationAccessToken().getAccessToken();
-        } else {
-            return acquireAadUserAccessToken().getAccessToken();
+    String acquireAccessToken() throws DataServiceException  {
+        try {
+            switch (authenticationType) {
+                case APP:
+                    return acquireAadApplicationAccessToken().getAccessToken();
+                case USER:
+                    return acquireAadUserAccessToken().getAccessToken();
+                case DEVICE:
+                    return acquireAccessTokenUsingDeviceCodeFlow().getAccessToken();
+                default:
+                    return acquireAccessTokenUsingDeviceCodeFlow().getAccessToken();
```

Comment:
Maybe we should throw an exception here?
or at least log.warn this since it doesn't made sense to reach the default...

Reply:
it is not supposed to.. No way it can, I think we saw it together, that the compiler forces to use default since null can be assigned to enum..Anyway, I think I will throw exception here, since device is not really default...